http/metrics.go

```
@@ -35,13 +35,37 @@ type Metrics struct {
 	circuitBreakerOpen  *prometheus.CounterVec
 }
 
+// registerCollector will register the passed collector
+func registerCollector(registry prometheus.Registerer, collector prometheus.Collector, mustRegister bool) prometheus.Collector {
+	if mustRegister {
+		registry.MustRegister(collector)
+	} else {
+		if err := registry.Register(collector); err != nil {
+			if are, ok := err.(prometheus.AlreadyRegisteredError); ok {
+				// metric has been registered before so use existing metric
+				return are.ExistingCollector
+			} else {
+				// Something else went wrong so panic
+				panic(err)
```

@greg-mclain 
This is a potentially breaking change.  When this happened previously, the logic would silently ignore this even though nothing had been registered.

---

I'm ok with that. I don't think we'll have issues and if we do they'll be easy to track down.